Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -199,64 +199,47 @@ private void PullerLoop(CancellationToken cancellationToken)
-                                while (rewinded == null)
+                            ChainedBlock rewinded = null;
+                            while (rewinded == null)
+                            {
+                                token.ThrowIfCancellationRequested();
+
+                                uint256 hash = this.UTXOSet.Rewind().GetAwaiter().GetResult();
+                                rewinded = this.Chain.GetBlock(hash);
+                                if (rewinded == null)
                                 {
-                                    token.ThrowIfCancellationRequested();
-
-                                    uint256 hash = this.UTXOSet.Rewind().GetAwaiter().GetResult();
-                                    rewinded = this.Chain.GetBlock(hash);
-                                    if (rewinded == null)
-                                    {
-                                        this.logger.LogTrace("Rewound to '{0}', which is still not a part of the current best chain, rewinding further.", hash);
-                                    }
+                                    this.logger.LogTrace("Rewound to '{0}', which is still not a part of the current best chain, rewinding further.", hash);
                                 }
+                            }
 
-                                this.Tip = rewinded;
-                                this.Puller.SetLocation(rewinded);
-                                this.chainState.HighestValidatedPoW = this.Tip;
-                                this.logger.LogInformation("Reorg detected, rewinding from '{0}' to '{1}'.", lastTip, this.Tip);
+                            this.Tip = rewinded;
+                            this.Puller.SetLocation(rewinded);
+                            this.chainState.HighestValidatedPoW = this.Tip;
+                            this.logger.LogInformation("Reorg detected, rewinding from '{0}' to '{1}'.", lastTip, this.Tip);
 
-                                continue;
-                            }
+                            continue;
                         }
                     }
-
-                    this.logger.LogTrace("Block received from puller.");
-                    this.AcceptBlock(blockValidationContext);
                 }
-                catch (Exception ex)
-                {
-                    if (ex is OperationCanceledException)
-                    {
-                        if (this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
-                            return;
-                    }
 
-                    // TODO Need to revisit unhandled exceptions in a way that any process can signal an exception has been
-                    // thrown so that the node and all the disposables can stop gracefully.
-                    this.logger.LogCritical(new EventId(0), ex, "Consensus loop at Tip:{0} unhandled exception {1}", this.Tip, ex.ToString());
-                    NLog.LogManager.Flush();
```

Comment:
do we need to put this in in the AsyncLoop exception handler?

Reply:
we don't because asyncloop crash just crashes the loop itself, not the whole app, flush is needed only when whole app crashes